### ISSUE TITLE
dnsIP in node-config.yaml is not correctly set when service network is altered

### ISSUE THREAD

**Issue by ivanthelad** [openshift/openshift-ansible#2091]
on installation the dnsIP in node-config.yaml  should be set accordingly if the "openshift_master_portal_net" property is set to something other than 172.30.0.0. 

If openshift is installed with a different service subnet then all service resolutions fail as the "172.30.0.1" is "hard coded" node-config.yaml

**Comment by sdodson** [comment]
That shouldn't actually be the current behavior, unless you've set openshift_dns_ip to that value. See https://github.com/openshift/openshift-ansible/blob/master/filter_plugins/openshift_node.py#L13-L39

Can you share your inventory? Also which version of the ansible code are you using?

**Comment by ivanthelad** [comment]
Ansible version is what is delivered by the rhel openshift channels. I don't have a example of the host file at the moment. I will need to spin up another environment first.

**Comment by sdodson** [comment]
@ivanthelad Can you please gather the RPM version? I cannot reproduce this with a version newer than openshift-ansible-3.0.97

**Comment by sdodson** [comment]
I actually intended to leave this comment on the other issue you opened. I haven't verified this particular issue yet.

**Comment by Rydekull** [comment]
I also hit this issue, sort of. Not exactly what ivanthelad describes, but almost. I changed the openshift_master_portal_net to something other then the default (also adjusted the node networks).

Once deployed, I noticed pods were unable to resolve kubernetes.default.svc.cluster.local. This was because the nameserver(kubernetes.default.svc.cluster.local = first address in service network) was missing in /etc/resolv.conf. 

This was resolved by setting openshift_dns_ip manually. 

However, it is also worth to note, where I differ from ivanthelad is that node-config.yaml did update, setting the dnsIP to the node's own IP. But this was not useful and made everything fail.
